README.yaml

```diff
@@ -0,0 +1,272 @@
+    url: "https://www.terraform.io/docs/registry/modules/publish.html#requirements"
+  - name: "Terraform Version Pinning"
+    description: "The required_version setting can be used to constrain which versions of the Terraform CLI can be used with your configuration."
+    url: "https://www.terraform.io/docs/configuration/terraform.html#specifying-a-required-terraform-version"
+  - name: "Terraform `templatefile` Function"
+    description: "`templatefile` reads the file at the given path and renders its content as a template using a supplied set of template variables."
+    url: "https://www.terraform.io/docs/configuration/functions/templatefile.html"
+  - name: "Terraform `template_file` data source"
+    description: "The `template_file` data source renders a template from a template string, which is usually loaded from an external file."
+    url: "https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file"
+  - name: "Deepmerge"
+    description: "Terraform module to perform a deep map merge of standard Terraform maps and objects."
+    url: "https://github.com/Imperative-Systems-Inc/terraform-modules/tree/master/deepmerge"
+
+# Short description of this project
+description: |-
+  Terraform module that accepts local or remote YAML stack configurations
+  and returns deep-merged variables, backend config, and remote state outputs for Terraform and helmfile components.
+
+# Introduction to the project
+introduction: |-
+
+  The module has three sub-modules:
+
+    - [vars](modules/vars) - accepts stack configuration and returns deep-merged variables for a Terraform or helmfile component.
+    - [backend](modules/backend) - accepts stack configuration and returns backend config for a Terraform component.
+    - [remote-state](modules/remote-state) - accepts stack configuration and returns remote state outputs for a Terraform component.
+      The module supports `s3` and `remote` (Terraform Cloud) backends.
+
+  ### Attributions
+
+  Big thanks to [Imperative Systems Inc.](https://github.com/Imperative-Systems-Inc)
+  for the excellent [deepmerge](https://github.com/Imperative-Systems-Inc/terraform-modules/tree/master/deepmerge) Terraform module
+  to perform a deep map merge of standard Terraform maps and objects.
+
+# How to use this module. Should be an easy example to copy and paste.
+usage: |-
+  For a complete example, see [examples/complete](examples/complete).
+
+  For an example on how to configure remote state for Terraform components in YAML config files and then read the components outputs from the remote state,
+  see [examples/remote-state](examples/remote-state).
+
+  For automated tests of the complete example using [bats](https://github.com/bats-core/bats-core) and [Terratest](https://github.com/gruntwork-io/terratest),
+  see [test](test).
+
+
+# Example usage
+examples: |-
+
```

Comment:
Here's an example stack configuration file:


```yaml
import:
    - ue2-globals

  vars:
    stage: dev

  terraform:
    vars: {}

  helmfile:
    vars: {}

  components:
    terraform:
      vpc:
        command: "/usr/bin/terraform-0.13"
        backend:
          s3:
            workspace_key_prefix: "vpc"
        vars:
          cidr_block: "10.102.0.0/18"
      eks:
        backend:
          s3:
            workspace_key_prefix: "eks"
        vars:

    helmfile:
      nginx-ingress:
        vars:
          installed: true
```

The `import` section refers to other stack configurations that are automatically deep merged.